From abf002af766c7f7c5e9ecb6eba4dc12b08a8f1d8 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Caol=C3=A1n=20McNamara?= <caolanm@redhat.com>
Date: Wed, 8 Jun 2016 11:35:15 +0100
Subject: [PATCH] Resolves: rhbz#1343766 a11y queries during dispose trigger
 combobox crashes

Change-Id: I29863ca95e64ccd31795f78c29c89ff35009d718
(cherry picked from commit f120abb446bf3f5230ed06a3b148654dde36bb94)
(cherry picked from commit f7b0cbf1df98a1c42fe25cfdbd9977fba78347c8)
---
 vcl/source/control/combobox.cxx | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/vcl/source/control/combobox.cxx b/vcl/source/control/combobox.cxx
index 94dc649..8abef91 100644
--- a/vcl/source/control/combobox.cxx
+++ b/vcl/source/control/combobox.cxx
@@ -1029,11 +1029,15 @@ OUString ComboBox::GetEntry( sal_Int32 nPos ) const
 
 sal_Int32 ComboBox::GetEntryCount() const
 {
+    if (!m_pImpl->m_pImplLB)
+        return 0;
     return m_pImpl->m_pImplLB->GetEntryList()->GetEntryCount() - m_pImpl->m_pImplLB->GetEntryList()->GetMRUCount();
 }
 
 bool ComboBox::IsTravelSelect() const
 {
+    if (!m_pImpl->m_pImplLB)
+        return false;
     return m_pImpl->m_pImplLB->IsTravelSelect();
 }
 
@@ -1052,6 +1056,8 @@ void ComboBox::EnableMultiSelection( bool bMulti )
 
 bool ComboBox::IsMultiSelectionEnabled() const
 {
+    if (!m_pImpl->m_pImplLB)
+        return false;
     return m_pImpl->m_pImplLB->IsMultiSelectionEnabled();
 }
 
-- 
2.7.3

